Content.Server/GameObjects/Components/Movement/ClimbableComponent.cs

```diff
@@ -0,0 +1,223 @@
+﻿
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.Components;
+using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using Robust.Server.Interfaces.Player;
+using Content.Server.Interfaces;
+using Content.Shared.GameObjects.EntitySystems;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Content.Shared.GameObjects.Components.Movement;
+using Content.Shared.Interfaces;
+using Content.Server.GameObjects.Components.Body;
+using System;
+
+namespace Content.Server.GameObjects.Components.Movement
+{
+    [RegisterComponent]
+    public class ClimbableComponent : SharedClimbableComponent, IDragDropOn
+    {
+#pragma warning disable 649
+        [Dependency] private readonly IServerNotifyManager _notifyManager = default!;
+        [Dependency] private readonly IPlayerManager _playerManager = default!;
+#pragma warning restore 649
+
+        /// <summary>
+        ///     The range from which this entity can be climbed.
+        /// </summary>
+        [ViewVariables]
+        private float _range;
+
+        /// <summary>
+        ///     The time it takes to climb the entity.
+        /// </summary>
+        [ViewVariables]
+        private float _climbDelay;
+
+        private ICollidableComponent _collidableComponent;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            _collidableComponent = Owner.GetComponent<ICollidableComponent>();
+        }
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref _range, "range", SharedInteractionSystem.InteractionRange / 1.4f);
+            serializer.DataField(ref _climbDelay, "delay", 0.8f);
+        }
+
+        bool IDragDropOn.CanDragDropOn(DragDropEventArgs eventArgs)
+        {
+            if (!ActionBlockerSystem.CanInteract(eventArgs.User))
+            {
+                _notifyManager.PopupMessage(eventArgs.User, eventArgs.User, Loc.GetString("You can't do that!"));
+
+                return false;
+            }
+
+            if (eventArgs.User == eventArgs.Dropped) // user is dragging themselves onto a climbable
+            {
+                if (eventArgs.Target == null) 
```

Comment:
Could this ever be null?

Reply:
I wasn't sure so i added the sanity check. In theory you should never be dragging to a null entity though so maybe it's overkill.